pkg/virt-api/rest/streamer_test.go

```diff
@@ -370,6 +374,12 @@ var _ = Describe("Streamer", func() {
 		Eventually(streamToServerCalled, defaultTestTimeout).Should(Receive())
 	})
 	It("closes the result channel after both streams have returned", func() {
+		// This test is mutually exclusive with the `-race` flag, as there is no other way to check
```

Comment:
> Isn't it enough to do
> 
> ```
> var results chan streamFuncResult
> ```
> 
> ?

Unfortunately not, the whole point of giving something a one-way channel is to force a specific use case for it. As such it makes sense that it is not possible to convert a one-way channel back to 2-way.
```
cannot use result (variable of type chan<- streamFuncResult) as chan streamFuncResult value in assignment
```
I did consider changing the function itself to receive a 2-way channel. However there is a reason it is restricted to write-only and i'm not convinced changing it such that we would need to constantly be aware to only write to the channel is worth having the testcase run during `make go-test`, especially considering that most of the CI is using `make bazel-test` instead.

Reply:
Do you mean something like
```
type streamFuncResultChannel struct {
    Result chan StreamFuncResult
    Open bool
}
```
?

I can do that, but the more i look into the topic, the more i ask myself how important it is that the channel is closed.
Since the test actually demonstrates that the channel is only given as write-only, which panics when writing to a closed channel, it does not make that much sense to close it in the first place.
The only thing closing a channel does is signaling readers that no new data is coming. Contrary to what the established design is, our reader is actually the one closing the channel, which should be done by the writer.

So since the resources are reclaimed by the garbage collector anyway, regardless of if the channel is closed or not, i would rather propose something else:
1. We create 2 channels, one for each needed function
2. The function as writer is responsible for closing the channel
3. We drop the test entirely

Or is there something i am missing which requires to ensure that the channel is closed?